### ISSUE TITLE
Feature request: make error messages more actionable

### ISSUE THREAD

**Issue by ant0nsc** [Microsoft/CNTK#274]
I'm trying to run the AlexNet example on my own couple of images. I basically just took the AlexNet config (attached), put in my own mapping file, and ran it through CNTK. As a result, I get the exception: "invalid stoi argument" - this gives me pretty much no guidance where I should start digging. It would be great if error handling could be changed such that users get a better indication as to where the problem really lies.

<<<<<<<<<<<<<<<<<<<< PROCESSED CONFIG WITH ALL VARIABLES RESOLVED <<<<<<<<<<<<<<<<<<<<
command: Train AddTop5Eval Test 
precision = float
CNTKModelPath: c:/dev/main/private/CNTK/Example/Output/Models
CNTKCommandTrainInfo: Train : 90
CNTKCommandTrainInfo: CNTKNoMoreCommands_Total : 90
CNTKCommandTrainBegin: Train
NDLBuilder Using CPU
EXCEPTION occurred: invalid stoi argument
Usage: cntk configFile=yourConfigFile
For detailed information please consult the CNTK book

[example.txt](https://github.com/Microsoft/CNTK/files/186285/example.txt)

**Comment by frankseide** [comment]
Thanks! I don't see this error string in our source code, and there is no call stack, so this error seems to be thrown by of a CRT library. Looking at the code, this error seems to come out of the function that parses the ImageReader's mapFile, which seems to be specified by a parameter called "file," and which expects class labels to be numeric. Could you verify that they are? Let me also paste the piece of code below for reference. Please would you mind letting us know whether this information could solve your problem.

At the same, I will pass this on internally to get the error fixed.
```
        std::string classId;
        if (!std::getline(ss, imagePath, '\t') || !std::getline(ss, classId, '\t'))
        {
            RuntimeError("Invalid map file format, must contain 2 tab-delimited columns: %s, line: %d.",
                         mapPath.c_str(),
                         static_cast<int>(lineIndex));
        }

        description.m_id = lineIndex;
        description.m_chunkId = lineIndex;
        description.m_path = imagePath;
        description.m_classId = std::stoi(classId);
```

**Comment by ldmtwo** [comment]
Regardless, I wanted to make a request like this too. There are many vague error messages. A quick fix would be to report out the values that caused the error, but a better fix would to tell us where in the NDL, CNTK, or MEL this error may have occurred. Last week and previous weeks, I was seeing errors like "convolution dimensions are not valid". And I kept asking myself, which of these did it mean?  With just 1 or 2 numbers, I could have at least tried to calculate where the error was or what I may have been doing wrong.

**Comment by ant0nsc** [comment]
It turned out that I had an extra tab in the map file, leaving the class ID to be an empty string. I've just sent you a pull request with a  fix: #281
